packages/matchbox/src/components/Panel/Header.js

```diff
@@ -1,45 +1,49 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { buttonsFrom } from '../Button';
-import styled from 'styled-components';
-import { actions, header, headerText } from './styles';
-import { padding } from 'styled-system';
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { Column } from '../Column';
+import { Columns } from '../Columns';
+import { getChild, excludeChild } from '../../helpers/children';
 import { PanelPaddingContext } from './context';
-import { createPropTypes } from '@styled-system/prop-types';
 
-const HeaderOuter = styled('div')`
-  ${header}
-  ${padding}
-`;
-
-export const HeaderText = styled('div')`
-  ${headerText}
-`;
-
-const Actions = styled('div')`
-  ${actions}
-`;
-
-const actionOverrides = { flat: true, size: 'small' };
-
-const Header = ({ title, actions, p, ...rest }) => {
+const Header = React.forwardRef(function Header(props, userRef) {
+  const { borderBottom, children, className } = props;
+  const actions = getChild('Panel.Action', children);
+  const title = excludeChild(['Panel.Action'], children);
   const paddingContext = React.useContext(PanelPaddingContext);
 
-  const actionMarkup =
-    actions && actions.length ? <Actions>{buttonsFrom(actions, actionOverrides)}</Actions> : null;
-
   return (
-    <HeaderOuter {...paddingContext} {...rest} pb="0">
-      <HeaderText>{title}</HeaderText>
-      {actionMarkup}
-    </HeaderOuter>
+    <Box
+      borderBottom={borderBottom ? '400' : 'none'}
+      className={className}
+      p="500"
+      pb={borderBottom ? null : '0'} // Null to inherit p="500"
+      {...paddingContext}
+      ref={userRef}
+      tabIndex="-1"
+    >
+      <Columns collapseBelow="xs" space="300">
```

Comment:
Action buttons look a little off on mobile when collapsed because of button padding. Not sure if we want to do something about that or not. 
<img width="259" alt="Screen Shot 2020-08-03 at 11 52 14 AM" src="https://user-images.githubusercontent.com/3878251/89207108-dc314200-d57f-11ea-82e4-bb6338878cc5.png">